libAvKys/Plugins/Crop/src/cropelement.h

```diff
@@ -0,0 +1,90 @@
+/* Webcamoid, webcam capture application. Crop Plug-in.
+ * Copyright (C) 2022  Tj <hacker@iam.tj>
+ *
+ * Webcamoid is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Webcamoid is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Webcamoid. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Web-Site: http://webcamoid.github.io/
+ */
+
+#ifndef CROPELEMENT_H
+#define CROPELEMENT_H
+
+// QRectF (not QRect) needed to interface with QML which uses real (float/double) values
+#include <QRectF>
+#include <akelement.h>
+
+class CropElementPrivate;
+
+class CropElement: public AkElement
+{
+    Q_OBJECT
+    Q_DISABLE_COPY(CropElement)
+    Q_PROPERTY(QRectF box
+               READ box
+               WRITE setBox
+               RESET resetBox
+               NOTIFY boxChanged)
+    Q_PROPERTY(QRectF minimum
+               READ minimum
+               WRITE setMinimum
+               RESET resetMinimum
+               NOTIFY minimumChanged)
+    // represents size of original frame; not changable externally
```

Comment:
If the properties are not supposed to be used externally, don't make them public, private things must be moved to the private class. Keep public just the minimal code to use the class.

Reply:
This is deliberate in order to allow additional control from the GUI if requested.

Reply:
I should add that both `minimum` and `maximum` are required for `NOTIFY` and, if the QML functionality is modified, `READ`.